scripts/jenkins/red_hat/red_hat_docker_certification.py

```diff
@@ -22,10 +22,18 @@ def has_scan_errors(digest, pid, api_key):
     response = None
     while response is None:
         intermediate_response = requests.get(url=url, headers=headers)
+        print("Intermediate response: {}".format(intermediate_response.status_code))
         if intermediate_response.status_code == 200:
-            response = intermediate_response
+            try:
+                intermediate_response.json()["data"]["results"]
+                response = intermediate_response
+            except (KeyError, TypeError):
```

Comment:
Sometimes, the body is missing when this is hit right after the scan is completed. Therefore, those errors are expected. If there are no results in the body, yet the response is `200`, this is considered to be a failure and the API is polled again. After a while, the API will start to return proper response according to documentation.

Reply:
what about checking for the key presence instead of catching the error and then logging the response contents?

Reply:
also, the comment is a good indicator that this should be moved to a function

Reply:
The data might be missing, or just the results might be missing. Or the results might be empty, resulting in the TypeError. This code is like Pokemon - gotta catch them good. I think it's good enough, but sure won't help to traverse it manually.

Agreed with the function, why not.